src/core/ext/filters/http/message_compress/message_compress_filter.c

```diff
@@ -159,7 +174,17 @@ static grpc_error *process_send_initial_metadata(
   error = grpc_metadata_batch_add_tail(
       exec_ctx, initial_metadata, &calld->accept_encoding_storage,
       GRPC_MDELEM_ACCEPT_ENCODING_FOR_ALGORITHMS(
-          channeld->supported_compression_algorithms));
+          channeld->supported_compression_algorithms &
+          GRPC_MESSAGE_COMPRESSION_ALGORITHM_MASK));
```

Comment:
please create a separate enum for stream compression. There's too much (unnecessary IMHO) hackery to reuses the single enum.